Monika-After-Story/MonikaModDev
# Rose reaction kiss

# **Additions:**

#### `persistent._mas_last_kiss`:

Contains a `datetime.datetime` of the last time you received a kiss from Monika

# **Changes:**

Rose gift reaction was changed to include a kiss at the end of reaction if certain conditions are met.

# **Testing**

- Verify that `persistent._mas_last_kiss` is set properly after every kiss.
- Verify that you get no kiss after gifting roses unless the following conditions are met:
  1. Enamored+ aff
  2. Already got your first kiss
  3. Random chance that depends on if you gave her roses before and if it's a special day
  4. At least one hour from the last kiss

- Verify no crashes ~~or random combustion of the gifted roses.~~
